server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1924,6 +1924,7 @@ public void onFailure(Exception e) {
 
         private void recordTelemetry() {
             usageService.getCcsUsageHolder().updateUsage(usageBuilder.build());
+            usageBuilder.setRemotesCount(0); // prevent double recording
```

Comment:
Hmm, I don't think I understand this. Did you see a case of double recording? Since this builder is an instance var of SearchResponseActionListener what scenario would cause recordTelemetry to be called twice/double?